jenkins/scripts/cbts/rules/base.py

```
@@ -0,0 +1,58 @@
+      - `needs_diff_for`: file paths / glob patterns whose diffs this rule consumes
+                         (Groovy uses this to decide which files to fetch diffs for)
+
+    Subclasses implement `apply(pr)` returning either None (not applicable) or
+    a RuleResult.
+    """
+
+    name: str = ""
+    needs_diff_for: list[str] = []
+
+    @abstractmethod
+    def apply(self, pr: PRInputs) -> Optional[RuleResult]: ...
```

Flake8 (7.3.0)

[error] multiple statements on one line (def)

(E704)